### ISSUE TITLE
isOpen method hiding exception messages

### ISSUE THREAD

**Issue by tebs** [sensiolabs/BehatPageObjectExtension#57]
Hi, 

I'm using the isOpen method (from SensioLabs\Behat\PageObjectExtension\PageObject\Page class) like this:

``` php
PHPUnit_Framework_Assert::assertTrue(
            $this->getPage('Home page')->isOpen(),
            'Home page is not open'
        );
```

The problem is that when the page fails to open I have no clue why, just my generic message `'Home page is not open' `. This is because isOpen method is catching all the exceptions thrown during verification (and their useful messages). 

I did a workaround and overrode the isOpen method without the try/catch, but, from my point of view Page public interface should provide this. 

What do you think? 

Thanks!

**Comment by jakzal** [comment]
I usually implement an after step hook, which outputs html on failed step (or part of it, or a screenshot if driver supports it). Would a built in support for this kind of thing solve your problem?

**Comment by tebs** [comment]
Well, it would definitely help for most of the cases. In my case, what was 'failing' was that the url was different after attempting to submit a form with non valid data, so the wonderful message thrown by the verifyUrl method was lost. 

I did a workaround by overriding the isOpen method in my BasePage class as follows:

```php
    public function isOpen(array $urlParameters = array())
    {
        $this->verify($urlParameters);

        return true;
    }
```
So I don't lost the exception messages. 

I checked the history of that method and at the beginning it didn't catch the exceptions but was changed to return a boolean value.

**Comment by PWalkow** [comment]
@jakzal Yes, build in support would be the best idea rather than adding this step manually to all projects and/or base pages.
